core-bundle/src/Resources/contao/pages/PageRegular.php

```diff
@@ -176,11 +176,8 @@ protected function prepare($objPage)
 
 		$this->Template->sections = $arrCustomSections;
 
-		// Mark RTL languages (see #7171)
-		if ($GLOBALS['TL_LANG']['MSC']['textDirection'] == 'rtl')
-		{
-			$this->Template->isRTL = true;
-		}
+		// Mark RTL languages (see #7171, #3360)
+		$this->Template->isRTL = 'rtl' === ((\ResourceBundle::create($GLOBALS['TL_LANGUAGE'], 'ICUDATA', true)['layout']['characters'] ?? null) === 'right-to-left' ? 'rtl' : 'ltr');
```

Comment:
```suggestion
		// Mark RTL languages (see #7171, #3360)
 		if ((\ResourceBundle::create($GLOBALS['TL_LANGUAGE'], 'ICUDATA', true)['layout']['characters'] ?? null) == 'right-to-left')
 		{
 			$this->Template->isRTL = true;
 		}
```

Reply:
Accessing `$GLOBALS['TL_LANGUAGE']` here seems odd? Why not the request locale?

Reply:
Actually we already have the appropriate locale available in a variable at this point. Changed in c48467c.